apps/src/lib/node/ledger/shell/mod.rs

```diff
@@ -691,10 +721,67 @@ where
                 }
             }
         }
-
         response
     }
 
+    /// If a handle to an Ethereum oracle was provided to the [`Shell`], attempt
+    /// to signal it to start, using an initial configuration based on
+    /// Ethereum bridge parameters in blockchain storage.
+    fn ensure_ethereum_oracle_started(&mut self) {
+        if let ShellMode::Validator {
+            eth_oracle: Some(EthereumOracleChannels { control_sender, .. }),
+            eth_oracle_started: ethereum_oracle_started,
+            ..
+        } = &mut self.mode
+        {
+            if *ethereum_oracle_started {
+                return;
+            }
+            let config = match EthereumBridgeConfig::read(&self.storage) {
+                Some(config) => config,
+                None => {
+                    tracing::warn!(
+                        "An Ethereum oracle task appears to be running, but \
+                         there are no Ethereum bridge parameters configured \
+                         in block storage, so this oracle will do nothing"
+                    );
+                    return;
+                }
+            };
```

Comment:
if no config is present, we will spam logs with this warning, every commit. is this the desired behavior?

Reply:
oh yeah good catch, we don't want to spam the logs - I've replaced this with a comment but now need to double check this is correct - if the Ethereum bridge will be launched by hard fork, then configuration will always be written during `InitChain` (so before any block is committed). In which case, validators should turn the oracle on depending on whether the Ethereum bridge is configured for the chain, and if they turn the oracle on when it's not needed, maybe that should be a misconfiguration error for which we could just stop the node